tests/integration/pilot/outboundtrafficpolicy/helper.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package outboundtrafficpolicy
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/apps"
+	"istio.io/istio/pkg/test/framework/components/galley"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/components/pilot"
+	"istio.io/istio/pkg/test/util/retry"
+)
+
+const (
+	IstioSystemScope = `
+apiVersion: networking.istio.io/v1alpha3
+kind: Sidecar
+metadata:
+ name: restrict-to-istio-system
+spec:
+ egress:
+ - hosts:
+   - "istio-system/*"
+`
+)
+
+// This test checks external traffic calls behave as expected. Rather than call real external endpoints,
+// we will restrict the config scope with a Sidecar resource, such that the client doesn't know
+// about the destination app, allowing it to follow the same path as a standard external call.
+func ExternalRequest(expectedCodes []string, t *testing.T) {
+	framework.
+		NewTest(t).
+		Run(func(ctx framework.TestContext) {
+			g := galley.NewOrFail(t, ctx, galley.Config{})
+			pilotInstance := pilot.NewOrFail(t, ctx, pilot.Config{Galley: g})
+
+			ns, err := namespace.New(ctx, "apps", true)
+			if err != nil {
+				t.Fatalf("failed to create namespace: %v", err)
+			}
+
+			// We want to test "external" traffic. To do this without actually hitting an external endpoint,
+			// we can import only istio-system so the destination app is not known to Envoy.
+			if err := g.ApplyConfig(ns, IstioSystemScope); err != nil {
+				t.Errorf("failed to set sidecar: %v", err)
+			}
+
+			instance := apps.NewOrFail(ctx, t, apps.Config{
+				Pilot: pilotInstance,
+				AppParams: []apps.AppParam{
+					{Name: "client"},
+					{Name: "dest"},
+				},
+				Namespace: ns,
+			})
+
+			client := instance.GetAppOrFail("client", t)
+			dest := instance.GetAppOrFail("dest", t)
+
+			for _, ep := range dest.EndpointsForProtocol(model.ProtocolHTTP) {
+				retry.UntilSuccessOrFail(t, func() error {
+					resp, _ := client.Call(ep, apps.AppCallOptions{})
```

Comment:
This doesn't look right. Either Call should return an error, on which you take an recompensating action, or it returns a success value with err as nil. Mixing and matching is confusing.

If the current Call method's signature is not good enough, we can refactor it, or add a new one that better suites your needs.